ducktape/tests/reporter.py

```diff
@@ -79,22 +80,26 @@ def report(self):
 
 
 class SingleResultFileReporter(SingleResultReporter):
+    def __init__(self, datum, results_dir):
+        super(SingleResultFileReporter, self).__init__(datum)
+        self.results_dir = results_dir
+
     def report(self):
         self.width = DEFAULT_SEPARATOR_WIDTH
-        report_file = os.path.join(self.result.test_context.results_dir, "report.txt")
+        report_file = os.path.join(self.results_dir, "report.txt")
         with open(report_file, "w") as fp:
             fp.write(self.report_string())
 
-        # write collected data
```

Comment:
I don't particularly like the extra clutter, but is there any chance someone will miss having the output JSON readily available in the output directory? Is it easy enough to get to otherwise, either via the `ResultStore` or in some log, that it won't be missed?